packages/fastify-vite/types/index.d.ts

```diff
@@ -18,12 +18,55 @@ declare module 'fastify' {
 
 type FastifyVitePlugin = FastifyPluginAsync<NonNullable<fastifyVite.FastifyViteOptions>>;
 
+type RouteType = Partial<{
+  server: unknown
+  req: unknown
+  reply: unknown
+  head: unknown
+  state: unknown
+  data: Record<string, unknown>
+  firstRender: boolean
+  layout: unknown
+  getMeta: unknown
+  getData: unknown
+  onEnter: unknown
+  streaming: unknown
+  clientOnly: unknown
+  serverOnly: unknown
+}>
+type Loosen<T> = T & Record<string, unknown>
+// eslint-disable-next-line @typescript-eslint/no-redundant-type-constituents
+type Ctx = Loosen<{ routes: Array<RouteType>, context: unknown, body: unknown, stream: unknown, data: unknown }>
+
+interface RendererFunctions {
+  createHtmlTemplateFunction(source: string): unknown
+  createHtmlFunction(source: string, scope?: unknown, config?: unknown): (ctx: Ctx) => Promise<unknown>;
+  createRenderFunction(args: Loosen<{ routes: Array<RouteType>, create: (arg0: Record<string, unknown>) => unknown, createApp: unknown }>): Promise<(server: unknown, req: unknown, reply: unknown) => Ctx | { element:string, hydration:string }>
+}
+
+// eslint-disable-next-line @typescript-eslint/no-redundant-type-constituents
+interface RendererOption<ClientModuleType = string | Record<string, unknown> | unknown, ClientType = unknown> extends RendererFunctions {
+  clientModule: ClientModuleType
+  createErrorHandler(client: ClientType, scope: FastifyInstance, config?: unknown): (error: Error, req?: FastifyRequest, reply?: FastifyReply) => void
+  createRoute(args: Loosen<{ client?: ClientType, handler?: (...args: unknown[]) => unknown, errorHandler: (error: Error, req?: FastifyRequest, reply?: FastifyReply) => void, route?: RouteType }>, scope: FastifyInstance, config: unknown): void
+  createRouteHandler(client: ClientType, scope: FastifyInstance, config?: unknown): (req: FastifyRequest, reply: FastifyReply) => Promise<unknown>
+  prepareClient(clientModule: ClientModuleType, scope?: FastifyInstance, config?: unknown): Promise<ClientType>
```

Comment:
Any chance we break any line over 80 columns, i.e., list those properties on their own lines?